test/SlowTests/Tests/Linq/CanQueryAndIncludeRevisions.cs

```diff
@@ -0,0 +1,1982 @@
+        {
+            using (var store = GetDocumentStore())
+            {
+                const string id = "users/omer";
+
+                await RevisionsHelper.SetupRevisions(Server.ServerStore, store.Database);
+
+                using (var session = store.OpenAsyncSession())
+                {
+                    await session.StoreAsync(new User
+                        {
+                            Name = "Omer",
+                        },
+                        id);
+
+                    await session.SaveChangesAsync();
+                }
+
+                string changeVector;
+                using (var session = store.OpenAsyncSession())
+                {
+                    var metadatas = await session.Advanced.Revisions.GetMetadataForAsync(id);
+                    Assert.Equal(1, metadatas.Count);
+
+                    changeVector = metadatas.First().GetString(Constants.Documents.Metadata.ChangeVector);
+
+                    session.Advanced.Patch<User, string>(id, x => x.ChangeVector, changeVector);
+                    await session.SaveChangesAsync();
+                }
+
+                using (var session = store.OpenAsyncSession())
+                {
+              
+                    var query = await session.Advanced
+                        .AsyncRawQuery<User>("from Users as u where u.Name = 'Omer' include revisions($p0)")
+                        .AddParameter("p0", "ChangeVector")
+                        .ToListAsync();
+
+                    var revision = await session.Advanced.Revisions.GetAsync<User>(changeVector);
+
+                    Assert.NotNull(revision);
+                    Assert.Equal(1, session.Advanced.NumberOfRequests);
+                }
+            }
+        }
+        
+        [Fact]
+        public void Query_RawQueryGetRevisionBeforeDateTime()
+        {
+            using (var store = GetDocumentStore())
+            {
+                const string id = "users/omer";
+                
+                 RevisionsHelper.SetupRevisions(Server.ServerStore, store.Database).Wait();
+        
+                using (var session = store.OpenSession())
+                {
+                     session.Store(new User
+                        {
+                            Name = "Omer",
+                        },
+                        id);
+
+                     session.SaveChanges();
+                }
+                using (var session = store.OpenSession())
+                {
+                    var metadatas =  session.Advanced.Revisions.GetMetadataFor(id);
+                    Assert.Equal(1, metadatas.Count);
+
+                    var changeVector = metadatas.First().GetString(Constants.Documents.Metadata.ChangeVector);
+                    var getRevisionsBefore = DateTime.UtcNow;
+                    var query =  session.Advanced
+                        .RawQuery<User>("from Users as u include revisions($p0)")
```

Comment:
In here, we have basically:

* `revisions(string)` which is a property in the document
* `revisions(date)` which is revision before, right?

I think it would be better to have different names.